sdk/src/cache/messages/data/scalar/get_batch.rs

```diff
@@ -0,0 +1,224 @@
+use momento_protos::cache_client::ECacheResult;
+use std::{
+    collections::HashMap,
+    convert::{TryFrom, TryInto},
+};
+
+use crate::{
+    cache::MomentoRequest,
+    utils::{self, parse_string},
+    CacheClient, IntoBytesIterable, MomentoError, MomentoResult,
+};
+
+use crate::cache::messages::data::scalar::get::{GetResponse, Value};
+
+/// Request to get a batch of items from a Momento Cache
+///
+/// # Arguments
+///
+/// * `cache_name` - name of cache
+/// * `keys` - list of keys to fetch
+///
+/// # Examples
+/// Assumes that a CacheClient named `cache_client` has been created and is available.
+/// ```
+/// # fn main() -> anyhow::Result<()> {
+/// # use momento_test_util::create_doctest_cache_client;
+/// # tokio_test::block_on(async {
+/// # let (cache_client, cache_name) = create_doctest_cache_client();
+/// use std::convert::TryInto;
+/// use std::collections::HashMap;
+/// use momento::cache::{GetBatchRequest, GetResponse};
+/// # cache_client.set(&cache_name, "key1", "value1").await?;
+/// # cache_client.set(&cache_name, "key2", "value2").await?;
+///
+/// let get_batch_request = GetBatchRequest::new(cache_name, vec!["key1", "key2"]);
+/// let results_map: HashMap<String, GetResponse> = cache_client.send_request(get_batch_request).await?.into();
+/// # assert_eq!(results_map.clone().len(), 2);
+///
+/// for (key, response) in results_map {
+///     match response {
+///         GetResponse::Hit { value } => {
+///             let value: String = value.try_into().expect("I stored a string!");
+///             println!("Fetched value for key {}: {}", key, value);
+///         }
+///         GetResponse::Miss => println!("Cache miss for key {}", key),
+///     }
+/// }
+/// # Ok(())
+/// # })
+/// # }
+/// ```
+pub struct GetBatchRequest<K: IntoBytesIterable> {
+    cache_name: String,
+    keys: K,
+}
+
+impl<K: IntoBytesIterable> GetBatchRequest<K> {
+    /// Constructs a new GetBatchRequest.
+    pub fn new(cache_name: impl Into<String>, keys: K) -> Self {
+        Self {
+            cache_name: cache_name.into(),
+            keys,
+        }
+    }
+}
+
+impl<K: IntoBytesIterable> MomentoRequest for GetBatchRequest<K> {
+    type Response = GetBatchResponse;
+
+    async fn send(self, cache_client: &CacheClient) -> MomentoResult<GetBatchResponse> {
+        // consume self.keys once to convert all keys to bytes
+        let byte_keys: Vec<Vec<u8>> = self.keys.into_bytes();
+
+        // convert keys to strings for the HashMap keys
+        let string_keys: Vec<String> = byte_keys
+            .iter()
+            .map(|key| parse_string(key.clone()))
+            .collect::<MomentoResult<Vec<String>>>()?;
+
+        let get_requests = byte_keys
+            .iter()
+            .map(|key| momento_protos::cache_client::GetRequest {
+                cache_key: key.clone(),
```

Comment:
you don't need a memcpy here for these strings
```suggestion
            .into_iter()
            .map(|key| momento_protos::cache_client::GetRequest {
                cache_key: key,
```